### ISSUE TITLE
Edit file functionality doesn't load for a specific file

### ISSUE THREAD

**Issue by zhzy0077** [go-gitea/gitea#29408]
### Description

Clicking "Edit file" in web portal takes forever and never loads for a specific file. It works pretty well for all other files.

**However, git cli works perfectly for the file.**

### Gitea Version

1.21.6 built with GNU Make 4.4.1, go1.21.7 : bindata, timetzdata, sqlite, sqlite_unlock_notify

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist
<scrubbed>

### Screenshots
<scrubbed>

### Git Version

git version 2.40.1

### Operating System

Linux

### How are you running Gitea?

docker compose (with a few env configurations removed - mostly db related)
```
version: "3.8"
services:
  gitea:
    image: gitea/gitea
    container_name: gitea
    restart: unless-stopped
    ports:
      - 52222:22
    volumes:
      - /root/gitea:/data
      - /etc/timezone:/etc/timezone:ro
      - /etc/localtime:/etc/localtime:ro
      - /mnt/data/backup/gitea:/gitea-backup
    networks:
      - production
networks:
  production:
    external: true
```

### Database

MySQL/MariaDB

**Comment by wxiaoguang** [comment]
Please use Gitea 1.21 nightly, it should have been fixed there.

**Comment by lunny** [comment]
Fixed by #29370 and will be released on v1.21.7

**Comment by zhzy0077** [comment]
It works after upgrading (temporarily) to 1.21-nightly. Thanks for the prompt answers!
